ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateLiteralConstantFix.fs

```diff
@@ -0,0 +1,90 @@
+
+open FSharp.Compiler.Symbols
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+
+type UpdateLiteralConstantFix(error: LiteralConstantValuesDifferError) =
+    inherit FSharpQuickFixBase()
+    let errorRefPat = error.Pat.As<IReferencePat>()
+    
+    let tryFindSigFile (topRefPat: ITopReferencePat) =
+        let containing = topRefPat.GetContainingTypeDeclaration()
+        let decls = containing.DeclaredElement.GetDeclarations()
+        decls |> Seq.tryFind (fun d -> d.GetSourceFile().IsFSharpSignatureFile)
+        
+    let tryFindSigBindingSignature sigMembers =
+        sigMembers
+        |>  Seq.tryPick(fun m ->
+            let bindingSignature = m.As<IBindingSignature>()
+            match bindingSignature with
+            | null -> None
+            | _ ->
+                match error.Pat.Binding.HeadPattern with
+                | :? IReferencePat as implPat ->
+                    match bindingSignature.HeadPattern with
+                    | :? IReferencePat as sigRefPat when
+                        implPat.DeclaredName = sigRefPat.DeclaredName -> Some bindingSignature
+                    | _ -> None
+                | _ -> None
+            )
+
+    let isField =
+        let refExpr = error.Pat.Binding.Expression.As<IReferenceExpr>()
+        isNotNull refExpr && refExpr.Reference.GetFcsSymbol() :? FSharpField
+            
+    let mutable sigRefPat = null
+
+    override x.Text = $"Update literal constant {error.Pat.Identifier.Name} in signature"
+
+    override x.IsAvailable _ =
+        // Todo reuse/extend SignatureFixUtil
+        if isNotNull errorRefPat then
+            match tryFindSigFile error.Pat with
+            | Some sigFile ->
+                let sigMembers = sigFile.As<IModuleDeclaration>().Members
+                let sigBindingSignature = tryFindSigBindingSignature sigMembers
+                match sigBindingSignature with
+                | None -> false
+                | Some s ->
+                    match s.HeadPattern with
+                    | :? IReferencePat as sRefPat ->
+                        sigRefPat <- sRefPat
+                        
+                        if isField then
+                            let implSymbolUse = errorRefPat.GetFcsSymbolUse()
+                            let implMfv = implSymbolUse.Symbol :?> FSharpMemberOrFunctionOrValue
+                            sRefPat.CheckerService.ResolveNameAtLocation(
+                                sRefPat, [ implMfv.ReturnParameter.Type.TypeDefinition.DisplayName ], false, null)
+                            |> Option.isSome
+                        elif error.Pat.Binding.Expression :? IReferenceExpr then
+                            let exprText = error.Pat.Binding.Expression.GetText()
```

Comment:
The same question about the pattern. The expression node access is also duplicated here.

Reply:
changed